talismane_core/src/main/java/com/joliciel/talismane/parser/ParserRegexBasedCorpusReader.java

```diff
@@ -228,9 +230,34 @@ public boolean hasNextConfiguration() {
 							}
 
 							if (!badConfig) {
-								PretokenisedSequence tokenSequence = new PretokenisedSequence(talismaneSession);
+								Sentence sentence = null;
+								if (sentenceReader != null && sentenceReader.hasNextSentence()) {
+									sentence = new Sentence(sentenceReader.nextSentence(), talismaneSession);
+								} else {
+									LinguisticRules rules = talismaneSession.getLinguisticRules();
+									if (rules == null)
+										throw new TalismaneException("Linguistic rules have not been set.");
+
+									String text = "";
+									for (ParseDataLine dataLine : dataLines) {
```

Comment:
`dataLines` will be as long as a single sentence - so we can take Proust as a typical maximum size. Initially considered using a StringBuilder, but `rules.shouldAddSpace(text, word)` requires a String, and if we change the required argument to a CharSequence, we lose out on convenience methods such as endsWith (which should have been implemented by CharSequence as a default method!)